intervals/interval.py

```diff
@@ -98,10 +98,11 @@ def wrapper(self, arg):
                 return func(self, arg)
             except IntervalException:
                 return NotImplemented
-        try:
-            arg = type(self)(self.type(arg))
-        except (ValueError, TypeError):
-            pass
+        if str(type(arg)) != "<class 'sqlalchemy.util.langhelpers._symbol'>":
```

Comment:
This is definitely not the right way to implement this. Now this implementation would be SQLAlchemy specific and not address any similar issues.

Reply:
Nice to know this project's alive :smile: 
I've been using this at work. Perhaps this is more in line with what should be done? I could wrap the sqlalchemy import in `try/catch`

```python
from functools import wraps

from intervals import interval
from sqlalchemy.orm.base import NO_VALUE


def overflow_patch(func):
    "https://github.com/kvesteri/intervals/issues/42"

    @wraps(func)
    def wrapper(self, arg):
        if arg == NO_VALUE:
            return func(self, None)
        return func(self, arg)

    return wrapper


def apply_patches():

    interval.AbstractInterval.__eq__ = overflow_patch(interval.AbstractInterval.__eq__)
    interval.AbstractInterval.__gt__ = overflow_patch(interval.AbstractInterval.__gt__)
    interval.AbstractInterval.__lt__ = overflow_patch(interval.AbstractInterval.__lt__)
    interval.AbstractInterval.__ge__ = overflow_patch(interval.AbstractInterval.__ge__)
    interval.AbstractInterval.__le__ = overflow_patch(interval.AbstractInterval.__le__)
    interval.AbstractInterval.__contains__ = overflow_patch(
        interval.AbstractInterval.__contains__
    )
    interval.AbstractInterval.__add__ = overflow_patch(
        interval.AbstractInterval.__add__
    )
    interval.AbstractInterval.__sub__ = overflow_patch(
        interval.AbstractInterval.__sub__
    )
    interval.AbstractInterval.glb = overflow_patch(interval.AbstractInterval.glb)
    interval.AbstractInterval.lub = overflow_patch(interval.AbstractInterval.lub)
    interval.AbstractInterval.inf = overflow_patch(interval.AbstractInterval.inf)
    interval.AbstractInterval.sup = overflow_patch(interval.AbstractInterval.sup)
    interval.AbstractInterval.__rsub__ = overflow_patch(
        interval.AbstractInterval.__rsub__
    )

```

Reply:
No, that is not the solution either. That would still impose dependency to SQLAlchemy which just doesn't make sense - this library should *NOT* have an SQLAlchemy dependency.

Reply:
I agree that sqlalchemy should not be a dependency for this lib. While a new version is not being released for this lib, perhaps a patch to the latest version could contain something similar to this so that people can continue to use it?

What I was proposing in the previous comment had no need to add a dependency on sqlalchemy. Something like this:

```python

try:
  from sqlalchemy.orm.base import NO_VALUE
  patch_no_value = True
except ImportError:
  patch_no_value = False

## Later in the code where the break occurs we can do:

if patch_no_value and arg == NO_VALUE:
    arg = None
```